lib/engine/step/g_1817/buy_sell_par_shares.rb

```diff
@@ -175,11 +175,26 @@ def process_bid(action)
           end
         end
 
+        def available_company_options(entity)
+          value = entity.companies.map(&:value)
+          # Rather naive way of doing this, but the number of potential values is small
+          options = [0]
+          value.each do |v|
```

Comment:
are you trying to do value.combination ?

Reply:
values = entity.companies.map(&:values)
(1..values.size).flat_map { |size| values.combination(size).to_a }

Reply:
Yeah, I'm away until Monday so won't be able to pick it up. I'm tempted to rework a bit more of this so it's automatic if players have no choice. Would then do a migration for existing games